### ISSUE TITLE
Narrator reads all the actionable elements in Dialog

### ISSUE THREAD

**Issue by sergul** [microsoft/fluentui#14600]
### Environment Information

- **Package version(s)**: (7.123.1)
- **Browser and OS versions**: (Chromium Edge 84.0.522.61 and Windows 10)

### Describe the issue:

Once a Dialog is opened, Narrator reads out all the actionable elements of its content.

### Please provide a reproduction of the issue in a codepen:

1. Go to https://codepen.io/sergulo/pen/KKdoMBK using Chromium Edge browser
2. Turn on the screen reader / Narrator
3. Click the "Open Dialog" button to open the Dialog
4. As soon as the Dialog is opened make sure the screen reader's focus is within the Dialog area

#### Actual behavior:

Narrator announces all the actionable elements of the Dialog content

#### Expected behavior:

Narrator should read the Dialog's content until it arrives at the first actionable element and stop reading after that.

**Comment by khmakoto** [comment]
Hi @sergul, I've taken a look at your example and this definitely reproduces. I'll work on it as soon as possible. Thanks for filing the issue!

**Comment by mimi-msft** [comment]
Hey @khmakoto do you have an eta on this fix? We have an open accessibility bug due to this which we can't close until it is resolved? Thanks!

**Comment by katiearriagam** [comment]
Are there any updates on this bug? @khmakoto, @paulgildea

**Comment by khmakoto** [comment]
@smhigley do you know what might be happening here? I've been trying to drill down the source of the problem but I can't seem to find it.

**Comment by paulgildea** [comment]
@tomi-msft as FYI to see if we can still root cause this issue.

**Comment by FosterSamuel** [comment]
@paulgildea @tomi-msft Any update on this issue?

**Comment by smhigley** [comment]
@sergul / @m-i-m-i the entire dialog is read because it is has `role="alertdialog"`, which is a live region role, similar to `role="alert"` (here are links to the [spec for alertdialog](https://w3c.github.io/aria/#alertdialog) and [pattern information for alert dialogs](https://w3c.github.io/aria-practices/#alertdialog)).

In Fluent, the blocking dialog (when `isBlocking` is set to true) is an alertdialog, I'm guessing because it's intended to be rare, and only for cases where the dialog is intended to be more intrusive. I'm not sure if there should always be a one-to-one relationship between blocking and `alertdialog`, but some sort of correlation makes sense.

I think the options are:
1. don't use a blocking dialog if you don't also want the alert behavior
2. Fluent provides a way to choose dialog vs. alertdialog separately from blocking vs. non-blocking

I think @khmakoto  or @paulgildea could provide more thoughts on whether the second option is likely :)

**Comment by kpasagada** [comment]
@smhigley Are there any updates on this bug? I am still able to see it in the latest Fluent UI version.

**Comment by smhigley** [comment]
@kpasagada an `alertdialog` is always going to read its entire contents when opened -- that's by design, and its entire purpose is to function as an alert, in dialog form.

If that's not the behavior you're looking for, then the solution is to make sure you have `isBlocking: false` in the Dialog properties. That's probably the best approach in general -- blocking dialogs should be super rare. If you think you do have an example of a dialog that should be blocking but not an alertdialog, could you post it here? It might help us understand whether there's a good use case to split those out, or add another property to the Dialog API to control them separately.

**Comment by sergul** [comment]
Sorry for late reply. @smhigley thanks for the workaround solutions. But looks like I have "isBlocking: false" in the repro example as you suggested above. And Narrator still announces the whole content of the Dialog.
1. don't use a blocking dialog if you don't also want the alert behavior

Regarding this second option, could you please provide more details how can I make the dialog non-alert ?
2. Fluent provides a way to choose dialog vs. alertdialog separately from blocking vs. non-blocking

**Comment by smhigley** [comment]
@sergul now that it's not an alertdialog, I actually cannot reproduce the original bug. Narrator doesn't read the full dialog contents for me, only the dialog title & close button, as expected.

**Comment by sergul** [comment]
@smhigley here is a link to demo the behavior on my end using Edge + Narrator
https://microsoft-my.sharepoint.com/:v:/p/v-semik/EbBwTm4-KpROhHiE-5Witt0B6lNY9eAkeMhzex3ch6n8cA

And here is my Edge version:
![image](https://user-images.githubusercontent.com/2910631/111202908-a9bddd80-8581-11eb-8be5-38ec5bc889e4.png)

**Comment by kpasagada** [comment]
@smhigley I have a blocking Panel component with `role="dialog"` and the Narrator reads the entire content of the Panel. I tried a couple of different Fluent UI versions and it seems to consistently read the contents of the panel. I also tried using Narrator on the Fluent UI Panel examples on the website and it still reads the panel contents.

**Comment by smhigley** [comment]
@sergul / @kpasagada do you get the same Narrator behavior of it reading through dialog content on this example? https://w3c.github.io/aria-practices/examples/dialog-modal/dialog.html

If so, it's likely just how Narrator behaves on its default settings. Multiple screen readers default to reading the entire content of a new page, page section, or dialog on initial navigation.
